game-core/src/main/java/games/strategy/engine/data/Unit.java

```diff
@@ -86,22 +62,6 @@ public boolean isEquivalent(final Unit unit) {
         && hits == unit.getHits();
   }
 
-  @Override
-  public int hashCode() {
-    if (type == null || owner == null || id == null || this.getData() == null) {
-      final String text =
-          "Unit.toString() -> Possible java de-serialization error: "
-              + (type == null ? "Unit of UNKNOWN TYPE" : type.getName())
-              + " owned by "
-              + (owner == null ? "UNKNOWN OWNER" : owner.getName())
-              + " with id: "
-              + getId();
-      UnitDeserializationErrorLazyMessage.printError(text);
-      return 0;
-    }
-    return Objects.hashCode(id);
-  }
```

Comment:
This reminds me of something. I believe some time ago we had this issue where something would silently fail due to serialization errors.
I don't remember who exactly dealt with this issue. I believe @ssoloff might be the person that added the code to fail-fast.
Not sure if this would be useful here as well.